pkg/gadgets/prometheus/tracer/tracer.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2023 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build !withoutebpf
+
+package tracer
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	otelprometheus "go.opentelemetry.io/otel/exporters/prometheus"
+	sdkmetric "go.opentelemetry.io/otel/sdk/metric"
+
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets"
+	igprometheus "github.com/inspektor-gadget/inspektor-gadget/pkg/prometheus"
+)
+
+var hostRoot string
+
+func init() {
+	hostRoot = os.Getenv("HOST_ROOT")
+}
+
+type Tracer struct{}
+
+func (t *Tracer) Run(gadgetCtx gadgets.GadgetContext) error {
+	ctx := gadgetCtx.Context()
+
+	params := gadgetCtx.GadgetParams()
+	metricsConfig := params.Get(ParamConfig).AsString()
+	metricsPath := params.Get(ParamMetricsPath).AsString()
+	listenAddress := params.Get(ParamListenAddress).AsString()
+
+	config, err := igprometheus.ParseConfig(filepath.Join(hostRoot, metricsConfig))
```

Comment:
Good point. Yes, I think the file should be on the client machine and somehow we should transfer its contents to the remote machine. 

I implemented a solution for this in https://github.com/inspektor-gadget/inspektor-gadget/pull/1620/commits/61df4a663378c8dda2cb44718380f9ff4fd28147, what do you think about this approach?

@flyth

Reply:
We had a similar discussion in #1527 - I don't think the Params library should read files itself, as that might become a security issue pretty fast. Instead, what we discussed in the other PR was to use a prefix '@' to point at a local file and then transfer it transparently as string (compressed in that case) - the file is locally being read by the cobra integration (registry.go). IMHO we should use that behavior globally.

Reply:
Ah, yeah - it would probably make sense to wrap the params before calling 
```
flag := cmd.PersistentFlags().VarPF(p, p.Key, p.Alias, desc)
```

Something like [this](https://github.com/inspektor-gadget/inspektor-gadget/commit/778ec5e1e5144cd26d968559788dbe7956334c11) could hook into that, leaving the verification intact.

/cc @burak-ok

Reply:
Fixed by https://github.com/inspektor-gadget/inspektor-gadget/pull/1656. Validation is working fine too.